Elements/src/Spatial/AdaptiveGrid/AdaptiveGraphRoutingWeightModifier.cs

```diff
@@ -67,29 +100,159 @@ public WeightModifier AddPlanarWeightModifier(string name, Plane plane, double f
                     return Math.Abs(plane.SignedDistanceTo(a.Point)) < _grid.Tolerance &&
                            Math.Abs(plane.SignedDistanceTo(b.Point)) < _grid.Tolerance;
                 }),
-                factor);
+                factor,
+                group);
+            AddWeightModifier(modifier);
+            return modifier;
+        }
+
+        /// <summary>
+        /// Create WeightModifier that sets the factor on all edges
+        /// parallel(both points must be withing influence radius of the polyline) or intersecting with given polyline.
+        /// </summary>
+        /// <param name="name">Name of new WeightModifier.</param>
+        /// <param name="polyline">Polyline to check if edge is affected or intersected.</param>
+        /// <param name="factor">Factor of new WeightModifier.</param>
+        /// <param name="influenceDistance">Influence radius of polyline.</param>
+        /// <param name="is2D">Whether edge and polyline comparison should be considered in 2d or 3d.</param>
+        /// <param name="group">Group name of new WeightModifier.</param>
+        /// <returns>Created WeightModifier.</returns>
+        public WeightModifier AddPolylineWeightModifier(string name, Polyline polyline, double factor, double influenceDistance, bool is2D, string group = null)
+        {
+            RoutingHintLine hintLine;
+            try
+            {
+                hintLine = new RoutingHintLine(polyline, factor, influenceDistance, true, is2D);
+            }
+            catch
+            {
+                hintLine = null;
```

Comment:
what can result in us not being able to make a hint line?  should we notify the user somehow that this happened?